crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -54,11 +54,13 @@ impl AuthorityStorePruner {
         for object in perpetual_db
             .objects
             .multi_get(object_keys_to_prune.iter())?
+            .into_iter()
+            .flatten()
         {
-            if let Some(StoreData::IndirectObject(indirect_object)) =
-                object.map(|o| o.into_inner().data)
-            {
-                *indirect_objects.entry(indirect_object.digest).or_default() -= 1;
+            if let StoreObject::Value(obj) = object.into_inner() {
```

Comment:
note that pruner doesn't handle/remove tombstones
there's no regression in terms of used space(we didn't GC parent_sync table before)

There's a potential edge case related to keeping tombstones for wrapped objects, so IMO it's better to handle tombstone pruning in follow up PR and include slightly different group of reviewers